pycheribuild/projects/run_fvp.py

```diff
@@ -114,17 +120,19 @@ def execute_fvp(self, args: list, disk_image_path: Path = None, firmware_path: P
             self.run_cmd(base_cmd + self._plugin_args() + args, **kwargs)
 
     def get_fvp_revision(self):
-        revpath = "sw/ARM_Fast_Models_FVP_Morello/rev"
-        try:
-            if self.use_docker_container:
-                rev = self.run_cmd(["docker", "run", "-it", "--rm", self.container_name, "cat",
-                                    "/opt/FVP_Morello/" + revpath], capture_output=True,
-                                   run_in_pretend_mode=True).stdout
-                return int(rev.strip())
-            return int(self.read_file(self.install_dir / revpath))
-        except Exception as e:
-            self.warning("Could not determine FVP revision:", e)
-            return 0
+        if self._fvp_revision is None:
+            revpath = "sw/ARM_Fast_Models_FVP_Morello/rev"
+            try:
+                if self.use_docker_container:
+                    rev = self.run_cmd(["docker", "run", "-it", "--rm", self.container_name, "cat",
+                                        "/opt/FVP_Morello/" + revpath], capture_output=True,
+                                       run_in_pretend_mode=True).stdout
+                    self._fvp_revision = int(rev.strip())
+                self._fvp_revision = int(self.read_file(self.install_dir / revpath))
```

Comment:
This needs an else now that there is no longer an early return.

Much better than returning zero!

Reply:
Oops yes